### PR TITLE
Change assert_ checks to assertEquals/NotEquals/True/False...

### PR DESCRIPTION
**Pull request by ewongbb**
This is for ticket 2521.

### CODE REVIEW

File: master/buildbot/test/unit/test_changes_p4poller.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def do_test_poll_successful(self, **kwargs):
         self.add_p4_describe_result(3, encoded_p4change[3])
 
         # The first time, it just learns the change to start at.
-        self.assert_(self.changesource.last_change is None)
+        self.assertTrue(self.changesource.last_change is None)
```
Comment (rutsky) [inline master/buildbot/test/unit/test_changes_p4poller.py]: Can we use `self.assertIsNone` here?

File: master/buildbot/test/unit/test_process_build.py [hunk 2]
```diff
@@ -157,7 +157,7 @@ def testRunSuccessfulBuild(self):
         b.startBuild(FakeBuildStatus(), None, self.workerforbuilder)
 
         self.assertEqual(b.results, SUCCESS)
-        self.assert_(('startStep', (self.workerforbuilder.worker.conn,), {})
+        self.assertTrue(('startStep', (self.workerforbuilder.worker.conn,), {})
```
Comment (rutsky) [inline master/buildbot/test/unit/test_process_build.py]: Can we use `self.assertIn(..., ...)`?

File: master/buildbot/test/unit/test_steps_transfer.py [hunk 3]
```diff
@@ -866,8 +866,8 @@ def testBasic(self):
                 self.assertEquals(
                     data, json.dumps(dict(sourcestamps=[ss.asDict()], properties={'key1': 'value1'})))
                 break
-        else:
-            self.assert_(False, "No downloadFile command found")
+            else:
+                self.assertRaise(ValueError, "No downloadFile command found")
```
Comment (rutsky) [inline master/buildbot/test/unit/test_steps_transfer.py]: Is there `self.assertRaise` in `TestCase`?

I know about `self.assertRaises`, and in this place I think original code is more corresponds to `raise ValueError("No downloadFile command found")/`

**Comment by rutsky** [comment]
Modern versions of `unittest` module have convenient methods like [`assertIn`](https://docs.python.org/3.5/library/unittest.html#unittest.TestCase.assertIn) and [`assertIsNone`](https://docs.python.org/3.5/library/unittest.html#unittest.TestCase.assertIsNone), however I'm not sure is they were backported to Twisted's `TestCase` or is Twisted's `TestCase` has proper alternative.

File: master/buildbot/test/unit/test_steps_transfer.py (hunk 3 above)
Comment (rutsky) [inline master/buildbot/test/unit/test_steps_transfer.py]: Also you changed logic here, when moved `else` to `if`.

**Comment by tardyp** [comment]
twisted TestCase is a subclass of unittest TestCase..

File: master/buildbot/test/unit/test_steps_transfer.py [hunk 4]
```diff
@@ -866,8 +866,8 @@ def testBasic(self):
                 self.assertEquals(
                     data, json.dumps(dict(sourcestamps=[ss.asDict()], properties={'key1': 'value1'})))
                 break
-        else:
-            self.assert_(False, "No downloadFile command found")
+            else:
+                raise ValueError("No downloadFile command found")
```
Comment (rutsky) [inline master/buildbot/test/unit/test_steps_transfer.py]: Why do you move `else` here?

Comment (ewongbb) [inline master/buildbot/test/unit/test_steps_transfer.py]: got confused.  Also didn't realize there was the for...else: syntax.

File: master/buildbot/test/unit/test_changes_p4poller.py (hunk 1 above)
Comment (Stibbons) [inline master/buildbot/test/unit/test_changes_p4poller.py]: yes, it is advised

File: master/buildbot/test/unit/test_process_build.py [hunk 5]
```diff
@@ -157,8 +157,8 @@ def testRunSuccessfulBuild(self):
         b.startBuild(FakeBuildStatus(), None, self.workerforbuilder)
 
         self.assertEqual(b.results, SUCCESS)
-        self.assert_(('startStep', (self.workerforbuilder.worker.conn,), {})
-                     in step.method_calls)
+        self.assertIn(('startStep', (self.workerforbuilder.worker.conn,), {}),
```
Comment (Stibbons) [inline master/buildbot/test/unit/test_process_build.py]: This is a good candidates for refactor in TestCase base classe (we use a tiny abstract class between unittest.TestCase and our tests, so we can implement our how missing assert* methods)...

  self.assertMethodCalledWith(step, ('startStep', (self.workerforbuilder.worker.conn,), {}),

File: master/buildbot/test/unit/test_process_build.py [hunk 6]
```diff
@@ -385,11 +385,11 @@ def claim(owner, access):
...
         b.startBuild(FakeBuildStatus(), None, self.workerforbuilder)
 
-        self.assert_(('startStep', (self.workerforbuilder.worker.conn,), {})
-                     not in step.method_calls)
+        self.assertNotIn(('startStep', (self.workerforbuilder.worker.conn,), {}),
+                     step.method_calls)
         self.assertEquals(claimCount[0], 1)
-        self.assert_(b.currentStep is None)
-        self.assert_(b._acquiringLock is not None)
+        self.assertTrue(b.currentStep is None)
```
Comment (Stibbons) [inline master/buildbot/test/unit/test_process_build.py]: assertIsNone

File: master/buildbot/test/unit/test_process_build.py [hunk 7]
```diff
@@ -385,11 +385,11 @@ def claim(owner, access):
 
         b.startBuild(FakeBuildStatus(), None, self.workerforbuilder)
 
-        self.assert_(('startStep', (self.workerforbuilder.worker.conn,), {})
-                     not in step.method_calls)
+        self.assertNotIn(('startStep', (self.workerforbuilder.worker.conn,), {}),
+                     step.method_calls)
         self.assertEquals(claimCount[0], 1)
-        self.assert_(b.currentStep is None)
-        self.assert_(b._acquiringLock is not None)
```
Comment (Stibbons) [inline master/buildbot/test/unit/test_process_build.py]: assertIsNotNone

File: master/buildbot/test/unit/test_process_metrics.py [hunk 8]
```diff
@@ -182,13 +182,13 @@ def testReconfig(self):
         # enable log_interval
         new_config.metrics = dict(log_interval=10, periodic_interval=0)
         observer.reconfigServiceWithBuildbotConfig(new_config)
-        self.assert_(observer.log_task)
+        self.assertTrue(observer.log_task)
         self.assertEquals(observer.periodic_task, None)
```
Comment (Stibbons) [inline master/buildbot/test/unit/test_process_metrics.py]: assertIsNone

File: master/buildbot/test/unit/test_steps_transfer.py [hunk 9]
```diff
@@ -867,7 +867,7 @@ def testBasic(self):
                     data, json.dumps(dict(sourcestamps=[ss.asDict()], properties={'key1': 'value1'})))
                 break
         else:
-            self.assert_(False, "No downloadFile command found")
+            raise ValueError("No downloadFile command found")
```
Comment (Stibbons) [inline master/buildbot/test/unit/test_steps_transfer.py]: self.fail("No downloadFile command found")

File: master/buildbot/test/unit/test_process_build.py (hunk 6 above)
Comment (gracinet) [inline master/buildbot/test/unit/test_process_build.py]: assertIsNone appears with python 2.7, I think (easy to backport in base test class)

**Comment by gracinet** [comment]
By the way, `assertEquals` is deprecated in favor of `assertEqual`, if that matters
https://docs.python.org/2.7/library/unittest.html#deprecated-aliases

File: master/buildbot/test/unit/test_process_build.py (hunk 6 above)
Comment (ewongbb) [inline master/buildbot/test/unit/test_process_build.py]: yes.. 2.7.  adding assertIsNone (and similar asserts) fail in 2.6, which is why I returned to assertTrue.

Comment (Stibbons) [inline master/buildbot/test/unit/test_process_build.py]: monkey patch :)

Comment (tardyp) [inline master/buildbot/test/unit/test_process_build.py]: buildbot master is now python2.7 only, so You have assertIsNone assertRaises, and all that fancy stuff.
Note that even when we were compatible 2.6, we added some of them via monkey patching (we might want to remove those btw)

Comment (rutsky) [inline master/buildbot/test/unit/test_process_build.py]: Is it 2.7 only? I think we agreed to not break 2.6 compatibility in master without special need on one of the last meetings.

Comment (tardyp) [inline master/buildbot/test/unit/test_process_build.py]: indeed, sorry. I was confused by the python2,6 PR from @sa2ajj

**Comment by tardyp** [comment]
in this case, lets merge it.

**Pull request merged**